llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Passes/TileAndDecomposeWinogradInputPass.cpp

```diff
@@ -0,0 +1,246 @@
+            loc, minMap,
+            ValueRange{
+                getValueOrCreateConstantIndexOp(rewriter, loc, offsets[i])});
+      }
+    }
+    rewriter.setInsertionPointToStart(loopNest.loops.back().getBody());
+    auto tensorType = RankedTensorType::get(
+        SmallVector<int64_t>(2, ShapedType::kDynamicSize), elementType);
+    auto dynamicSlice = rewriter
+                            .create<tensor::ExtractSliceOp>(
+                                loc, tensorType, input, offsets, sizes, strides)
+                            .getResult();
+
+    // Copy input slice into zeroed padded scratch space
+    strides = SmallVector<OpFoldResult>(2, one);
+    offsets = SmallVector<OpFoldResult>(2, zero);
+    sizes = SmallVector<OpFoldResult>{sizes[1], sizes[2]};
+    Value zeroScratch = rewriter
+                            .create<linalg::FillOp>(loc, ValueRange{zeroF32},
+                                                    ValueRange{scratch})
+                            .result();
+    auto inputSlice =
+        rewriter
+            .create<tensor::InsertSliceOp>(loc, dynamicSlice, zeroScratch,
+                                           offsets, sizes, strides)
+            .getResult();
+
+    // Extract output slice
+    strides = SmallVector<OpFoldResult>(inputOp.getOutputOperandRank(), one);
+    offsets = SmallVector<OpFoldResult>(2, zero);
+    offsets.append(ivs.begin(), ivs.end());
+    sizes = SmallVector<OpFoldResult>(inputOp.getOutputOperandRank(), one);
+    sizes[0] = sizes[1] = inputTileSizeAttr;
+    tensorType = RankedTensorType::get(inputTileSquare, elementType);
+    Value iterArg = loopNest.loops.back().getRegionIterArg(0);
+    auto outputSlice =
+        rewriter
+            .create<tensor::ExtractSliceOp>(loc, tensorType, iterArg, offsets,
+                                            sizes, strides)
+            .getResult();
+
+    // Create computation
+    Value accumulator, result, AMatrix, BMatrix;
+    linalg::MatmulOp matmulOp;
+    for (int i = 0; i < 2; i++) {
+      accumulator = rewriter
+                        .create<linalg::FillOp>(loc, ValueRange{zeroF32},
+                                                ValueRange{outputSlice})
+                        .result();
+      if (i == 0) {
+        AMatrix = inputSlice;
+        BMatrix = BV;
+      } else {
+        AMatrix = BTV;
+        BMatrix = result;
+      }
+      matmulOp = rewriter.create<linalg::MatmulOp>(
+          loc, tensorType, ValueRange{AMatrix, BMatrix}, accumulator);
+      if (i == 0) {
+        matmulOp->setAttr(IREE::LinalgExt::Winograd::getWinogradAttrName(),
+                          rewriter.getStringAttr("I x B"));
+      } else {
+        matmulOp->setAttr(IREE::LinalgExt::Winograd::getWinogradAttrName(),
+                          rewriter.getStringAttr("B' x I x B"));
+      }
+      result = matmulOp.getResult(0);
+    }
+
+    // Insert results into output slice
+    auto updatedOutput = rewriter
```

Comment:
Use `Value` as the type instead of `auto` should let us avoid calling `.getResult()` and thus this skewed code formating?

Reply:
Thanks will try that